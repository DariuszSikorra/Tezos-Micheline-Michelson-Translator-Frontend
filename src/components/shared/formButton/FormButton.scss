.form-button {
    font-weight: 500;
    text-transform: uppercase;
    letter-spacing: 0.15;
    font-size: 14px;
    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.2), 0px 2px 2px rgba(0, 0, 0, 0.12), 0px 0px 2px rgba(0, 0, 0, 0.14);
    border-radius: 4px;
    padding: 5px;
    height: 40px;
    min-width: 100px;
    border: none;
    color: white;
    
    &:hover {
        cursor: pointer;
    }
}

.form-button_submit {
    background: #C90000;
}

.form-button_secondary {
    background: #787878;
}