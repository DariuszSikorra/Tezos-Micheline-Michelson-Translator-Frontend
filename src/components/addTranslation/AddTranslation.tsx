import React from "react";
import { Formik } from "formik";
import * as Yup from "yup";
import { useMappedState } from "redux-react-hook";
import FormTextarea from "../shared/textarea/FormTextarea";
import FormInput from "../shared/input/FormInput";
import { IState } from "../../store/translator/types";
import FormCodeDisplay from "../shared/formCodeDisplay/FormCodeDisplay";
import FormButton from "../shared/formButton/FormButton";
import { AddTranslationState, FormValues } from "./types";
import "./AddTranslation.scss";

const mapState = (state: IState) => ({
  translator: state.translator,
});

const validationSchema = Yup.object().shape({
  title: Yup.string().required("Title is required"),
  author: Yup.string(),
  email: Yup.string()
    .required("To get notification the email is needed.")
    .email("It's a wrong email address."),
  description: Yup.string(),
});

const AddTranslation = ({ setShowModal }: AddTranslationState) => {
  const { translator } = useMappedState(mapState);

  const submitForm = (values: FormValues) => {
    setShowModal(false);
  };

  return (
    <div className="add-translation">
      <Formik
        initialValues={{
          title: "",
          author: "",
          isEmail: "",
          email: "",
          description: "",
        }}
        validationSchema={validationSchema}
        onSubmit={submitForm}
      >
        {({ values, errors, touched, handleChange, handleBlur, handleSubmit, isSubmitting }) => (
          <form onSubmit={handleSubmit} noValidate>
            <div className="add-translation_fields-line">
              <FormInput
                label="Title"
                type="text"
                name="title"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.title}
                errors={errors.title}
                touched={touched.title}
              />
              <FormInput
                label="Author"
                type="text"
                name="author"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.author}
                errors={errors.author}
                touched={touched.author}
              />
            </div>
            <div className="add-translation_fields-line">
              <FormInput
                label="Email"
                type="email"
                name="email"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.email}
                errors={errors.email}
                touched={touched.email}
                className={!values.isEmail ? "blocked" : "undefined"}
                disabled={!values.isEmail}
              />
              <FormInput
                label="Send me a notification via email"
                type="checkbox"
                name="isEmail"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.isEmail}
                errors={errors.isEmail}
                touched={touched.isEmail}
                isValidationDisplay={false}
              />
            </div>
            <FormTextarea
              label="Description"
              placeholder="Fill in the description."
              name="description"
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.description}
              errors={errors.description}
              touched={touched.description}
            />
            <div className="add-translation_code-displays">
              <FormCodeDisplay value={translator.micheline} type="Micheline" />
              <FormCodeDisplay value={translator.michelson} type="Michelson" />
            </div>
            <div className="add-translation_buttons">
              <FormButton label="cancel" type="secondary" onClick={() => setShowModal(false)} />
              <FormButton
                label="save"
                type="submit"
                onClick={() => submitForm(values)}
                disabled={isSubmitting || !!errors}
              />
            </div>
          </form>
        )}
      </Formik>
    </div>
  );
};

export default AddTranslation;
