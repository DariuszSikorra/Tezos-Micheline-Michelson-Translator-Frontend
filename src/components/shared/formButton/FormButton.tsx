import React from "react";
import "./FormButton.scss";
import { ButtonProps } from "./types";

const FormButton = ({ type, label, className, disabled }: ButtonProps) => (
  <button
    className={`form-button form-button_${type || "submit"} ${className}`}
    data-testid="form-button"
    disabled={disabled}
    type="submit"
  >
    {label}
  </button>
);

export default FormButton;
