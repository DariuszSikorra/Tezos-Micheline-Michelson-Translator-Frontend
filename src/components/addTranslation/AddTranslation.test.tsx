import React from "react";
import { mount } from "enzyme/build";
import { StoreContext } from "redux-react-hook";
import AddTranslation from "./AddTranslation";
import FormButton from '../shared/formButton/FormButton';

import configureStore from "../../store/index";
import rootSaga from "../../store/saga";

const store = configureStore();
store.runSaga(rootSaga);

describe("AddTranslation", () => {
  const wrapper = mount(
    <StoreContext.Provider value={store}>
      <AddTranslation setShowModal={() => { }} />
    </StoreContext.Provider>,
  );

  it('matches snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const item = wrapper.find(AddTranslation);

    expect(
      item
        .at(0)
        .find({ name: "name" })
        .at(0).text()
    ).toEqual("Name");

    expect(
      item
        .at(0)
        .find(FormButton)
        .at(0)
        .text()
    ).toEqual("cancel");
  });
});
