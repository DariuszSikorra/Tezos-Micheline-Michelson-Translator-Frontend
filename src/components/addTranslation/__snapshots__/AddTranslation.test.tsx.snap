// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTranslation matches snapshot 1`] = `
"<AddTranslation setShowModal={[Function: setShowModal]}>
  <div className=\\"add-translation\\">
    <Formik initialValues={{...}} validationSchema={{...}} onSubmit={[Function: submitForm]} validateOnChange={true} validateOnBlur={true} isInitialValid={false} enableReinitialize={false}>
      <form onSubmit={[Function]} noValidate={true}>
        <div className=\\"add-translation_fields-line\\">
          <FormInput label=\\"Name\\" type=\\"text\\" name=\\"name\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]}>
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Name
              </p>
              <input name=\\"name\\" type=\\"text\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Name\\" value=\\"\\" test-id=\\"input-test\\" className=\\"form-input_input\\" />
              <div className=\\"form-input_validation-marker undefined\\" />
              <div className=\\"form-input_error\\" />
            </div>
          </FormInput>
          <FormInput label=\\"Author\\" type=\\"text\\" name=\\"author\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]}>
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Author
              </p>
              <input name=\\"author\\" type=\\"text\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Author\\" value=\\"\\" test-id=\\"input-test\\" className=\\"form-input_input\\" />
              <div className=\\"form-input_validation-marker undefined\\" />
              <div className=\\"form-input_error\\" />
            </div>
          </FormInput>
        </div>
        <FormTextarea label=\\"Description\\" placeholder=\\"Fill in the description.\\" name=\\"description\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]}>
          <div className=\\"form-textarea undefined\\">
            <div className=\\"form-textarea_label\\">
              Description
            </div>
            <textarea className=\\"form-textarea_input\\" placeholder=\\"Fill in the description.\\" name=\\"description\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" />
            <div className=\\"form-textarea_error\\" />
          </div>
        </FormTextarea>
        <div className=\\"add-translation_code-displays\\">
          <FormCodeDisplay value=\\"parameter int;\\\\nstorage int;\\\\ncode { CAR ;\\\\n       PUSH int 1 ;\\\\n       ADD ;\\\\n       NIL operation ;\\\\n       PAIR }\\" type=\\"Micheline\\">
            <div className=\\"form-code-display\\">
              <div className=\\"form-code-display_type\\">
                Micheline
              </div>
              <div className=\\"form-code-display_value\\">
                <pre>
                  parameter int;
                  storage int;
                  code { CAR ;
                         PUSH int 1 ;
                         ADD ;
                         NIL operation ;
                         PAIR }
                </pre>
              </div>
            </div>
          </FormCodeDisplay>
          <FormCodeDisplay value=\\"[ { \\"prim\\": \\"parameter\\", \\"args\\": [ { \\"prim\\": \\"int\\" } ] }, { \\"prim\\": \\"storage\\", \\"args\\": [ { \\"prim\\": \\"int\\" } ] }, { \\"prim\\": \\"code\\", \\"args\\": [ [ { \\"prim\\": \\"CAR\\" },{ \\"prim\\": \\"PUSH\\", \\"args\\": [ { \\"prim\\": \\"int\\" }, { \\"int\\": \\"1\\" } ] },{ \\"prim\\": \\"ADD\\" },{ \\"prim\\": \\"NIL\\", \\"args\\": [ { \\"prim\\": \\"operation\\" } ] },{ \\"prim\\": \\"PAIR\\" } ] ] } ]\\\\n\\" type=\\"Michelson\\">
            <div className=\\"form-code-display\\">
              <div className=\\"form-code-display_type\\">
                Michelson
              </div>
              <div className=\\"form-code-display_value\\">
                <pre>
                  [ { &quot;prim&quot;: &quot;parameter&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; } ] }, { &quot;prim&quot;: &quot;storage&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; } ] }, { &quot;prim&quot;: &quot;code&quot;, &quot;args&quot;: [ [ { &quot;prim&quot;: &quot;CAR&quot; },{ &quot;prim&quot;: &quot;PUSH&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;int&quot; }, { &quot;int&quot;: &quot;1&quot; } ] },{ &quot;prim&quot;: &quot;ADD&quot; },{ &quot;prim&quot;: &quot;NIL&quot;, &quot;args&quot;: [ { &quot;prim&quot;: &quot;operation&quot; } ] },{ &quot;prim&quot;: &quot;PAIR&quot; } ] ] } ]
                  
                </pre>
              </div>
            </div>
          </FormCodeDisplay>
        </div>
        <div className=\\"add-translation_buttons\\">
          <FormButton label=\\"cancel\\" type=\\"secondary\\" onClick={[Function: onClick]}>
            <button className=\\"form-button form-button_secondary undefined\\" data-testid=\\"form-button\\" onClick={[Function: onClick]} disabled={[undefined]}>
              cancel
            </button>
          </FormButton>
          <FormButton label=\\"save\\" type=\\"submit\\" onClick={[Function: onClick]} disabled={false}>
            <button className=\\"form-button form-button_submit undefined\\" data-testid=\\"form-button\\" onClick={[Function: onClick]} disabled={false}>
              save
            </button>
          </FormButton>
        </div>
      </form>
    </Formik>
  </div>
</AddTranslation>"
`;
